react-client/src/Components/AnnotationPanel/AddAnnotationPanel.js

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useEffect, useRef, useContext } from "react";
+import { v4 as uuidv4 } from "uuid";
+import { AiOutlineClose } from "react-icons/ai";
+import GlobalContext from "../../store/global-context";
+import useLocalStorageStore from "../../hooks/useLocalStorageStore";
+import * as domUtils from "../../utils/domUtils";
+
+
+const AddAnnotationPanel = React.memo(({onRemoveAnnotation, onUpdatedAnnotation, targetElement}) => {  
+  const [txtInput, setTxtInput] = useState("");
+  const [displayInput, setDisplayInput] = useState(false);
+  const [annotationStore, setAnnotationStore] = useLocalStorageStore('annotation', []);
+  const txtInputRef = useRef(null);
+  
+  useEffect(() => {
+    setTimeout(() => {
+      txtInputRef.current.focus();
+    }, 300);
+    return () => {};
+  }, []);
+  
+  useEffect(() => {      
+    const hasExistingAnnotations = domUtils.hasAnnotations(annotationStore, targetElement);
+    if (hasExistingAnnotations) {
+      const domIdentifier = domUtils.getUniqueElementIdentifierByTagAndIndex(targetElement);
+
+      let existingAnnotation = annotationStore.filter(
+        (obj) =>obj.domIndex == domIdentifier.index && obj.elem == domIdentifier.elType
+      );
+
+      setTxtInput(hasExistingAnnotations === true ? existingAnnotation[0].title : null);
```

Comment:
Redundant logic. No need to add: `hasExistingAnnotations === true ? existingAnnotation[0].title : null` since your logic is inside the if statement of `hasExistingAnnotations`.

Instead, do this: `setTxtInput(existingAnnotation[0].title);`

Reply:
yep line 31 implemented did not expect it would work that way was trying to prevent possible bugs if no data was found